ts/configuration/versions.ts

```diff
@@ -0,0 +1,13 @@
+import { API } from '..';
+
+import { ConfigurationV1 } from './v1';
+
+/**
+ * Provides access to the Kubernetes extension's Configuration API.
+ */
+export interface ConfigurationAPI {
```

Comment:
You also need to add a `configuration` attribute to the `ExtensionHelper` class.  This is what supports invoking the API via the `k8s.extension.foo.v3` syntax.  Also see `index.ts` where the library gets told about the lookup keys for the various interface categories.